subsys/power/device.c

```diff
@@ -66,6 +66,24 @@ int sys_pm_suspend_devices(void)
 	return 0;
 }
 
+int sys_pm_low_power_devices(void)
+{
+	for (int i = device_count - 1; i >= 0; i--) {
+		int idx = device_ordered_list[i];
+
+		device_retval[i] = device_set_power_state(&pm_device_list[idx],
+						DEVICE_PM_LOW_POWER_STATE,
+						NULL, NULL);
+		if (device_retval[i]) {
+			LOG_ERR("%s low power operation failed\n",
+					pm_device_list[idx].config->name);
+			return device_retval[i];
```

Comment:
see comment later - maybe we should reactivate devices powered down before the error occurred?